tts-voicerss/src/main/java/org/mobicents/media/server/tts/voicerss/VoiceRssEngine.java

```diff
@@ -0,0 +1,193 @@
+package org.mobicents.media.server.tts.voicerss;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.XMLConfiguration;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.NameValuePair;
+import org.apache.http.StatusLine;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.apache.log4j.Logger;
+import org.apache.shiro.crypto.hash.Sha256Hash;
+import org.mobicents.media.server.tts.core.SpeechSynthesizerException;
+import org.mobicents.media.server.tts.core.TTSEngine;
+
+
+/**
+ * VoiceRssEngine contains all the logic to parse the configuration file and processing text data
+ * 
+ * @author Thanh Tran
+ *
+ */
+public class VoiceRssEngine implements TTSEngine{
+	private final static Logger logger = Logger.getLogger(VoiceRssEngine.class);
+	private final static int BUFFER = 1024 * 8;
+	private static final List<NameValuePair> parameters;
+	static {
+		parameters = new ArrayList<NameValuePair>();
+		parameters.add(new BasicNameValuePair("c", "WAV"));
+		parameters.add(new BasicNameValuePair("f", "8khz_16bit_mono"));
+	}
+	
+	// Define elements of configuration xml file
+	private final static String 	SERVICE_ROOT = "service-root";
+	private final static String 	API_KEY = "apikey";
+	
+	// Configuration xml elements data
+	private String 					myAPIKey;
+	
+	private final URI myURIService;
+	private final Map<String, String> myMapLanguage;
+	
+	public VoiceRssEngine(final String configurationPath) throws ConfigurationException {
```

Comment:
The TTS Engine should receive the list of parameters already. It's not his job to load and parse values from configuration!